Sources/WalletConnectRelay/Dispatching.swift

```diff
@@ -84,28 +113,38 @@ final class Dispatcher: NSObject, Dispatching {
     }
 
     func connect() throws {
-        try socketConnectionHandler.handleConnect()
+        try socketConnectionHandler?.handleConnect()
     }
 
     func disconnect(closeCode: URLSessionWebSocketTask.CloseCode) throws {
-        try socketConnectionHandler.handleDisconnect(closeCode: closeCode)
+        try socketConnectionHandler?.handleDisconnect(closeCode: closeCode)
+    }
+    
+    private func setUpSocketConnection() {
+        setUpWebSocketSession()
+        setUpSocketConnectionObserving()
     }
 
     private func setUpWebSocketSession() {
-        socket.onText = { [unowned self] in
+        socket?.onText = { [unowned self] in
             self.onMessage?($0)
         }
     }
 
     private func setUpSocketConnectionObserving() {
-        socket.onConnect = { [unowned self] in
+        socket?.onConnect = { [unowned self] in
             self.socketConnectionStatusPublisherSubject.send(.connected)
         }
-        socket.onDisconnect = { [unowned self] _ in
+        socket?.onDisconnect = { [unowned self] error in
             self.socketConnectionStatusPublisherSubject.send(.disconnected)
-
+            if error != nil {
+                self.socket?.request.url = relayUrlFactory.create(
+                    host: relayHost,
+                    projectId: projectId
```

Comment:
could we maybe add host and projectId to the url factory init? they wan't change anyway